src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.revrobotics.CANSparkBase.SoftLimitDirection;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxSim;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.stateManagement.RobotStateManager;
+import frc.robot.utilities.DebugEntry;
+import frc.robot.utilities.LimelightHelpers;
+import frc.robot.utilities.TunableNumber;
+import java.util.function.Consumer;
+
+public class TurretSubsystem extends SubsystemBase {
+
+  private CANSparkMaxSim turretMotor;
+  private SingleJointedArmSim turretSim;
+  private Mechanism2d turretMech;
+  private MechanismRoot2d root;
+  private MechanismLigament2d turretAngleSim;
+  private ShuffleboardTab turretTab = Shuffleboard.getTab("Turret Tab");
+
+  private PIDController turretPIDController;
+  private CANcoder m_encoder;
+  private double turretPosition;
+  private double turretVelocity;
+  private Consumer<Double> testPosition;
+  private DebugEntry<Double> turretPositionEntry =
+      new DebugEntry<Double>(turretPosition, "Position", this);
+  private DebugEntry<Double> turretVelocityEntry =
+      new DebugEntry<Double>(turretVelocity, "Velocity", this);
+  private TunableNumber tunableTestPosition =
+      new TunableNumber("Turret Test Position", 30, testPosition);
```

Comment:
TunableNumber seems to not work well right now. Suggest just using a Shuffleboard/Smartdashboard entry for now.